internal_event_handlers.go

```diff
@@ -628,28 +662,41 @@ func (weh *workflowExecutionEventHandlerImpl) handleWorkflowExecutionCancelReque
 	weh.cancelHandler()
 }
 
-// Currently handles only side effect markers
 func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 	eventID int64,
 	attributes *m.MarkerRecordedEventAttributes,
 ) error {
-	if attributes.GetMarkerName() != sideEffectMarkerName {
+	switch attributes.GetMarkerName() {
+	case sideEffectMarkerName:
+		dec := gob.NewDecoder(bytes.NewBuffer(attributes.GetDetails()))
```

Comment:
Why are you not using encoding interface on hostEnv to decode, so we can switch easily with a different encoder.